src/SwarmFit/SwarmFitter.cs

```diff
@@ -57,88 +64,90 @@ double GetError(double[] vars)
     public FitSolution Solve(int iterations = 1000)
     {
         Stopwatch sw = Stopwatch.StartNew();
+        Random rand = Rand;
 
         double[] bestGlobalPositions = new double[VariableCount];
         double bestGlobalError = double.MaxValue;
 
-        Particle[] particles = new Particle[NumParticles];
-        List<FitSolution> intermediateSolutions = [];
+        Span<Particle> particles = new Particle[NumParticles];
+        List<FitSolution>? intermediateSolutions = StoreIntermediateSolutions ? [] : null;
```

Comment:
Instead of unconditionally adding result history later on insider the loop, only do so when the user asked for this.